Sources/ClientRuntime/Networking/Http/CRT/CRTClientTLSOptions.swift

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+import AwsCommonRuntimeKit
+
+public struct CRTClientTLSOptions {
+
+    /// Path to the trust store certificate
+    public let certificatePath: String?
+
+    /// Filename of the trust store certificate file in main bundle
+    public let certificateFilename: String?
+
+    /// Name of key store file. ex. /path/file
+    public let keyStoreFilepath: String?
+
+    /// Password for the key store if required.
+    public let keyStorePassword: String?
+
+    /// Path to a private key
+    public let privateKeyFilepath: String?
+
+    /// Information is provided to use custom trust store
+    public var useSelfSignedCertificate: Bool {
+        return certificatePath != nil && certificateFilename != nil
+    }
+
+    /// Information is provided to use custom key store
+    public var useProvidedKeystore: Bool {
+        return (keyStoreFilepath != nil && keyStorePassword != nil) ||
+            (privateKeyFilepath != nil && useSelfSignedCertificate)
+    }
+
+    public init(
+        certificatePath: String? = nil,
+        certificateFilename: String? = nil, // .cer
+        keyStoreFilepath: String? = nil, // .p12 PEM
+        keyStorePassword: String? = nil,
+        privateKeyFilepath: String? = nil
+    ) {
+        self.certificatePath = certificatePath
+        self.certificateFilename = certificateFilename
+        self.keyStoreFilepath = keyStoreFilepath
+        self.keyStorePassword = keyStorePassword
+        self.privateKeyFilepath = privateKeyFilepath
+    }
+}
+
+extension CRTClientTLSOptions {
+    func resolveContext() -> TLSContext? {
+        // Provide default tls context (nil)
+        var tlsContext: TLSContext?
+
+        // Set CRT client tls options with .client mode
+        do {
+            // Set tlsOptions to default value
+            var tlsOptions = TLSContextOptions.makeDefault()
+
+            if self.useProvidedKeystore {
+                #if os(tvOS) || os(iOS) || os(watchOS) || os(macOS)
+                if let path = keyStoreFilepath, let password = keyStorePassword {
+                    tlsOptions = try .makeMTLS(pkcs12Path: path, password: password)
+                }
+                #else
+                if let certPath = certificatePath,
+                    let certFilename = certificateFilename,
+                    let privateKeyPath = privateKeyFilepath {
+                    let certFilepath = "\(certPath)/\(certFilename)"
+                    tlsOptions = try .makeMTLS(certificatePath: certFilepath, privateKeyPath: privateKeyPath)
```

Comment:
`makeMTLS(certificatePath, privateKeyPath)` is also available on MacOS. The current implementation would fall back into default `tlsOptions` if the user tries to setup `certFilepath` and `privateKeyPath` on MacOS. 
I'm not sure if that's on purpose or not. If so, we probably wanna add comments to document it.

Reply:
Which should be used for visionOS?

Reply:
Ran latest version by @xiazhvera who says it looks good!